grafana-plugin/e2e-tests/integrations/maintenanceMode.test.ts

```diff
@@ -13,36 +13,30 @@ import { goToOnCallPage } from '../utils/navigation';
 type MaintenanceModeType = 'Debug' | 'Maintenance';
 
 test.describe('maintenance mode works', () => {
-  test.slow(); // this test is doing a good amount of work, give it time
-
   const MAINTENANCE_DURATION = '1 hour';
   const REMAINING_TIME_TEXT = '59m left';
   const REMAINING_TIME_TOOLTIP_TEST_ID = 'maintenance-mode-remaining-time-tooltip';
 
   const createRoutedText = (escalationChainName: string): string =>
     `alert group assigned to route "default" with escalation chain "${escalationChainName}"`;
 
-  const _openIntegrationSettingsPopup = async (page: Page): Promise<Locator> => {
-    const integrationSettingsPopupElement = page.getByTestId('integration-settings-context-menu');
-    await integrationSettingsPopupElement.waitFor({ state: 'visible' });
+  const _openIntegrationSettingsPopup = async (page: Page, shouldDoubleClickSettingsIcon = false): Promise<void> => {
+    await page.waitForTimeout(2000);
+    const integrationSettingsPopupElement = page
+      .getByTestId('integration-settings-context-menu-wrapper')
+      .getByRole('img');
     await integrationSettingsPopupElement.click();
-    return integrationSettingsPopupElement;
+    if (shouldDoubleClickSettingsIcon) {
+      await integrationSettingsPopupElement.click();
+    }
   };
 
   const getRemainingTimeTooltip = (page: Page): Locator => page.getByTestId(REMAINING_TIME_TOOLTIP_TEST_ID);
 
   const enableMaintenanceMode = async (page: Page, mode: MaintenanceModeType): Promise<void> => {
-    const integrationSettingsPopupElement = await _openIntegrationSettingsPopup(page);
-    /**
-     * we need to click twice here, because adding the escalation chain route
-     * doesn't unfocus out of the select element after selecting an option
-     */
```

Comment:
_nit_ we might want to move this comment up to line 29 just to give future us an idea why we did this (since it's _somewhat_ hacky imo)